### ISSUE TITLE
Stile type

### ISSUE THREAD

**Issue by matkoniecz** [streetcomplete/StreetComplete#2749]
### General
Affected tag(s) to be modified/added: [stile](https://wiki.openstreetmap.org/wiki/Key:stile)
Question asked: **What type of stile is this?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose - https://wiki.openstreetmap.org/wiki/Talk:Tag:barrier%3Dstile#Is_stile_type_used_for_anything_or_potentially_useful.3F
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one) - https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dstile#Stile_details
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

> stile=*, often together with step_count=*, is generally of very high value & use. Different types of stiles can pose difficulties for older people and those with limited mobility. For instance a squeezer with step_count=1 will generally be readily accessible, a ladder stile may be far too difficult. I very recently searched for all stiles on a particular route in the Peak District so that we can try & avoid the difficult ones to enable my father to climb a hill. Material is also useful in some contexts (squeezers, stepovers): a stepover made of stone with a high step count is likely to be beyond someone like my Dad, whereas a wooden one with step count of 2 or less he can manage (albeit with a bit of help)

### Ideas for implementation

**Element selection:** `barrier=stile !stile`

**Metadata needed:** No

**Proposed GUI:** Image selector

**Comment by Cj-Malone** [comment]
I've been meaning to suggest this, I would recommend an option (or "other answer") to be able to re tag a feature as `barrier=kissing_gate`, a fair amount in my local area have recently been replaced with them. In England grandfathered stiles can be repaired and replaced, but all new ones must be more kissing gates, in an aim to be more accessible.

**Comment by westnordost** [comment]
doesn't even need to be an other answer. Could be shown as a regular answer option.

**Comment by matkoniecz** [comment]
@Cj-Malone If you are familiar with stiles: more feedback about how `step_count` is tagged on stiles would be useful. It can unblock one more quest that appears to be potentially useful.

See https://wiki.openstreetmap.org/wiki/Talk:Tag:barrier%3Dstile#Is_stile_type_used_for_anything_or_potentially_useful.3F

> Is there a consensus how it should be tagged - for example stile that has 2 steps on one side and 3 on another. Is it step_count=3 or step_count=5? Stile with 1 step on each side - is it step_count=1 or something else?

(also, if someone familiar with stiles visited this page - your feedback also would be welcomed)
(OSM Wiki talk page or discussion on talk-gb is the best place for responses)

**Comment by arrival-spring** [comment]
> doesn't even need to be an other answer. Could be shown as a regular answer option.

If the question is "What type of stile is this?" then I think that an answer of "kissing gate" would be quite confusing as a kissing gate is not a type of stile, so if it's an option it should be in the other answers section.

**Comment by matkoniecz** [comment]
> If the question is "What type of stile is this?" then I think that an answer of "kissing gate" would be quite confusing


It would be "Replaced by kissing gate" what I think would be clear (not tested so far)

**Comment by matkoniecz** [comment]
Preview:

![screen12](https://user-images.githubusercontent.com/899988/115127561-b649b180-9fd7-11eb-889b-f4d6639db53a.png)


From what I see it would be nice to have ability to do `changes.deleteAllTags()` - for use in "converted into a kissing gate" option.

Right now test implementation deletes `step_count` `wheelchair` `bicycle` - but many other tags could be tagged on it, and all should be deleted.

**Comment by DerDings** [comment]
> Right now test implementation deletes `step_count` `wheelchair` `bicycle` - but many other tags could be tagged on it, and all should be deleted.
> 
> 

`access` (generic, probably signed not physical), `layer`, `level`, `operator` etc. should not be deleted, so just deleting curated tags is the best way imo. Maybe add `material` and `stroller` to the list of tags to delete.

**Comment by westnordost** [comment]
"Converted into..." sounds like "The owner converted this stile into a kissing gate" and thus implies that it has been a stile before. Is this deliberate?

**Comment by westnordost** [comment]
There is no PR for that yet, but just as a comment for that: Does it make sense to exclude all nodes that are exclusively part of ways with `access=private`?

**Comment by matkoniecz** [comment]
> "Converted into..." sounds like "The owner converted this stile into a kissing gate" and thus implies that it has been a stile before. Is this deliberate?

Yes, as it would be added on `barrier=stile` without `stile=yes`

> Does it make sense to exclude all nodes that are exclusively part of ways with access=private?

Yes, as AFAIK many of them or an `access=private` `foot=designated`/`foot=permissive` (public right of way for pedestrians or similar term in UK law).

Not sure whether anyone bothered erecting or mapping them on fully private ways.

Code is at https://github.com/matkoniecz/Zazolc/commits/stile_type - PR not opened as it relies on barrier quest being merged.

**Comment by matkoniecz** [comment]
> access (generic, probably signed not physical), layer, level, operator etc. should not be deleted
> Maybe add material, height, width and stroller to the list of tags to delete.

Done. Also added `dog_gate`.
